code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -869,3 +869,59 @@
 		log_game("DYNAMIC: Not enough players volunteered for the ruleset [name] - [candidates.len] out of [required_candidates].")
 		return FALSE
 	return TRUE
+
+//////////////////////////////////////////////
+//                                          //
+//           NINJA      (GHOST)             //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/ninja
+	name = "Space Ninja"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_HEAVY
+	antag_flag = ROLE_NINJA
+	required_type = /mob/dead/observer
+	antag_datum = /datum/antagonist/ninja
+	enemy_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	required_enemies = list(2,2,2,2,2,2,2,2,2,2)
+	required_candidates = 1
+	weight = 3
+	cost = 9
+	minimum_players = 20
```

Comment:
I would rather you use `requirements`. every array index is 6+ players, starting from 0, and the values are the threat required to spawn. Ninja should be a high threat antag to increase rarity. having a flat curve will make it too common..
`requirements = list(100,100,100,90,70,60,50,50,50,40)` for example. maybe see if `REQUIREMENTS_VERY_HIGH_THREAT_NEEDED` fits, although probably too extreme there

Reply:
The default values seem pretty good to me `list(40,30,20,10,10,10,10,10,10,10)`. I don't think it needs increasing as no other midround event excluding wizard and nukies have it increased so it is inline with all other midround events. Ninja is definitely not a nukie/wizard level midround antag as its chaos extends only as far as where the ninja is and once it is killed the chaos caused by it is killed with it. At the same time I don't think it can justify lowering the requirements since it is definitely more impactful than a traitor spawning.
I think it should be fine at the default values, as all other events are using the default values; if it spawns too frequently then we can increase the cost.